map.go

```diff
@@ -65,6 +66,11 @@ type MapSpec struct {
 	// InnerMap is used as a template for ArrayOfMaps and HashOfMaps
 	InnerMap *MapSpec
 
+	// Extra trailing bytes found in the ELF map definition when using structs
+	// larger than libbpf's bpf_map_def. Must be empty before instantiating
+	// the MapSpec into a Map.
+	Extra bytes.Buffer
```

Comment:
Why not `[]byte`?

Reply:
I'm trying to figure out what `MapSpec.Copy` should do with `Extra`. For consistency it would be nice to copy the field but doing that with a `Buffer` while keeping the original read position doesn't seem straightforward. `[]byte` doesn't have that problem and has a smaller API surface and you can get a reader by using `bytes.NewReader(spec.Extra)`.

Reply:
Good point, I completely missed this initially. For our use case, re-initializing the Buffer on the destination and then `Write()`ing the remaining bytes should be sufficient. It won't work with `UnreadByte()` and friends, but oh well,

```go
	cpy.Extra = bytes.Buffer{}
	_, _ = cpy.Extra.Write(ms.Extra.Bytes())
```

Note that `cpy.Extra = bytes.NewBuffer(ms.Extra.Bytes())` retains `ms.Extra`'s backing array and is not safe to use.

Reply:
We've decided to go with a `bytes.Reader` instead.